src/webcomponents/helperfunctions/jsonToCol.tsx

```diff
@@ -0,0 +1,85 @@
+
+export const jsonToCol=(languages:any)=>{
+let datas:any = JSON.parse(languages);
+let dataArray :any= [];
+let keys:any = [];
+let columns:any = [];
+let columnsValues:any = [];
+let rows:any = [];
+let fullRowData:any = [];
+let columnTitle:any = [{ type: "text", title: "messageID", width: 150 }];
+if (datas) {
+  for (const d in datas) {
+    columnTitle.push({
+      type: "text",
+      title: `${d}`,
+      width: 150,
+      data: "nice",
+    });
+    dataArray.push(datas[d]);
+  }
+}
+if (dataArray) {
+  for (const d in dataArray) {
+    columns.push(dataArray[d]);
+  }
+  for (const d in dataArray[0]) {
+    keys.push(d);
+  }
+}
+if (columns) {
+  let i = 0;
+  for (const key in columns) {
+    for (const k in keys) {
+      let row = [];
+      for (let j = 0; j < columns.length; j++) {
+        if (Object.keys(columns[key])[i] == keys[k]) {
+          row.push({ cols: columns[j][keys[k]] });
+        }
+      }
+      if (row.length > 0) {
+        rows.push(row);
+      }
+      row = [];
+    }
+    i++;
+  }
+}
+let columsValues = [];
+if (rows) {
+  for (const col in rows) {
+    if (rows[col].length > 0) {
+      columsValues.push(rows[col]);
+    }
+  }
+}
+if (columsValues) {
+  let datas = [];
+  for (const value in columsValues) {
+    for (const columnNumber in columsValues[value]) {
+      datas.push(columsValues[value][columnNumber]["cols"]);
+    }
+    columnsValues.push(datas);
+    datas = [];
+  }
+}
+
+if (columnsValues) {
+  for (let key = 0; key < keys.length; key++) {
```

Comment:
In previous code review I had two comments - please improve the code - I will repeat them:


1. I beleive that you can simplify the code by using http://cryto.net/~joepie91/blog/2015/05/04/functional-programming-in-javascript-map-filter-reduce/
2. Each loop should be wrapped as a function to improve readability (the name of the functions helps understanding of the code).

Reply:
map doesn't work with object that why i used loops

Reply:
Can't use function around loops as there lots of things happing that will mess up the data. what I can do is comment the purpose of the loop